tests/loggers/test_all.py

```diff
@@ -343,6 +343,47 @@ def _test_logger_created_on_rank_zero_only(tmpdir, logger_class):
     assert trainer.state.finished, f"Training failed with {trainer.state}"
 
 
+class AllRankLogger(LightningLoggerBase):
+    exp = "test exp"
+
+    def experiment(self):
+        return self.exp
+
+    def log_metrics(self, _metric, _step):
+        pass
+
+    def version(self):
+        pass
+
+    def name(self):
+        return "AllRank"
+
+    def log_hyperparams(self, _params, *_args, **_kwargs):
+        pass
+
+
+class AllRanksLoggerCheck(Callback):
+    def on_train_batch_start(self, trainer, _pl_module, _batch, _batch_idx, _dataloader_idx):
+        assert trainer.logger.experiment() == AllRankLogger.exp
+
+
+@RunIf(skip_windows=True)
+def test_logger_created_on_all_ranks(tmpdir):
+    logger = AllRankLogger()
+    model = BoringModel()
```

Comment:
does this check that data from each rank is actually available in the logger? 

- you could extend the boring model and log the rank in the training step
- then in the all rank logger, you can locally store the logged metrics
- then the callback on fit end can then assert that the trainer's rank maps to the value inside of the logger to confirm that this does run on all ranks